pkg/render/dex.go

```diff
@@ -268,7 +267,8 @@ func (c *dexComponent) deployment() client.Object {
 							VolumeMounts: c.dexConfig.RequiredVolumeMounts(),
 						},
 					},
-					Volumes: c.dexConfig.RequiredVolumes(),
+					Volumes:  c.dexConfig.RequiredVolumes(),
+					Affinity: podaffinity.NewPodAntiAffinity(DexObjectName, DexNamespace),
```

Comment:
Same as previous comment, the affinity should only be specified when Replicas is > 1.